Userland/Libraries/LibSQL/Type.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021, Tim Flynn <trflynn89@pm.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashMap.h>
+#include <AK/String.h>
+
+namespace SQL {
+
+#define ENUMERATE_SQL_TYPES \
+    __ENUMERATE_SQL_TYPE("text", Text, String)                   \
+    __ENUMERATE_SQL_TYPE("int", Integer, int)                    \
+    __ENUMERATE_SQL_TYPE("float", Float, double)
+
+enum SQLType {
+#define __ENUMERATE_SQL_TYPE(name, type, impl) type,
+    ENUMERATE_SQL_TYPES
+#undef __ENUMERATE_SQL_TYPE
+};
```

Comment:
this looks weird

Reply:
Isn't this the pattern Andreas uses?

Reply:
yes, it is. not necessarily weird, but haven't looked at it in detail yet.

Reply:
similar, but not quite
see 
https://github.com/SerenityOS/serenity/blob/master/Kernel/API/Syscall.h#L31

so it would be
```cpp
#define ENUMERATE_SQL_TYPES(S)   \
     S("text", Text, String)                     \
     S("int", Integer, int)                        \
     S("float", Float, double)

enum SQLType {
#define __ENUMERATE_SQL_TYPE(name, type, impl) type,
    ENUMERATE_SQL_TYPES(__ENUMERATE_SQL_TYPE)
#undef __ENUMERATE_SQL_TYPE
};

```

Reply:
Ah OK will fix